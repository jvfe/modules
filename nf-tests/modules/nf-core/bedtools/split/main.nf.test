nextflow_process {

    name "Test Process BEDTOOLS_SPLIT"
    script "modules/nf-core/bedtools/split/main.nf"
    process "BEDTOOLS_SPLIT"
    config "nf-tests/modules/nf-core/bedtools/split/nextflow.config"

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:

            }
            process {
                """
                input[0] = [
                             [ id:'test', scatter:2 ], // meta map
                             file(params.test_data['homo_sapiens']['genome']['genome_multi_interval_bed'], checkIfExists: true)
                           ]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
