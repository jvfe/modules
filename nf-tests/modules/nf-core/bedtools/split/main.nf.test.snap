{
    "Should run without failures": {
        "content": [
            {
                "beds": [
                    [
                        {
                            "id": "test",
                            "scatter": 2
                        },
                        [
                            "test.00001.bed:md5,d58e5e46c2fcc3b8be5db0f023e93cb5",
                            "test.00002.bed:md5,03caf952e9297a54620d2bbba8dc2823"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,b3de37d944970e1b8dd9f85b8fb2c1a2"
                ]
            }
        ],
        "timestamp": "2022-11-15T15:09:58+0000"
    }
}